File: 1. Two Sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        diff_map = {}
        for idx, val in enumerate(nums):
            diff = target - val
            if diff in diff_map:
                result.append(idx)
                result.append(diff_map[diff])
                break
            else:
                diff_map[val] = idx
        return result

File: 1. Two Sum/solution.scala
object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        var indexMap = Map[Int, Int]()
        for (i <- nums.indices) {
            val c = target - nums(i)
            if (indexMap.contains(c)) {
                return Array(indexMap(c), i)
            }
            indexMap += (nums(i) -> i)
        }
        return Array()
    }
}